docs/xks/developer-guide/container-security.md

```diff
@@ -0,0 +1,293 @@
+        capabilities:
+          drop:
+            - NET_RAW
+            - CAP_SYS_ADMIN
+      volumeMounts:
+        - mountPath: /tmp
+          name: tmp
+  volumes:
+    - emptyDir: {}
+      name: tmp
+```
+
+### Allow Privilege Escalation
+
+Disabling privilege escalation for all Pods is critical for applications to to gain greater access to the Node than they need to. This setting will probably never affect most applications.
+
+### Dropped Capabilities
+
+Linux capabilities are certain root permissions that can be given to containers without giving root access. Certain capabilities should never be given to a container as they would give too much access, for this reason they are always dropped.
+
+* `CAP_SYS_ADMIN` - Is the same as making the container root, would allow a container to escalate privileges.
+* `NET_RAW` - Allows a container to craft raw network packets which can be exploited for malicious actions.
+
+## Pod Constraints
+
+In addition to the default security context there are certain configuration settings which are not allowed in XKF. They have a lower impact on end users as they are opted into, but they are good to be aware of and can explain why a Pods creation may be blocked by the cluster. The enforcement is done before the Pod is created, this means that a Deployment may be allowed to be created but the Pods that result will not, causing a Deployment to always have zero replicas.
+
+### Host Namespaces
+
+Being able to share process IDs or memory between a container and a host means that the container would in theory have access to the host or any containers also running on the host. Setting `hostPID` or `hostIPC` to `true` would allow for this to happen. For this reason that is not allowed.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  hostPID: true
+  hostIPC: true
+```
+
+### Host Networking
+
+Allowing a Pod access to the host network namespace would mean giving it access to the hosts loopback device and enable it to listen to the network traffic of all Pods on the host.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  hostNetwork: true
+```
+
+Using a host port for a Pods container will result in binding that hosts port to the container port. This would expose the container to the network outside of the cluster. This can cause issues in cases where the ports have to be unique, and is generally a feature that is not needed in XKF which is my it is not allowed.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  containers:
+    - ports:
+        - hostPort: 10902
+```
+
+### Privileged Containers
+
+Privileged containers have root capabilities on the host machine and are there for not allowed in XKF. There would be no separation between the container and the host if this setting were to be allowed.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: app
+spec:
+  containers:
+    - securityContext:
+        privileged: true
```

Comment:
I'm 99% sure this should be false and not true.
```
➜ k explain pod.spec.containers.securityContext.privileged  
KIND:     Pod
VERSION:  v1

FIELD:    privileged <boolean>

DESCRIPTION:
     Run container in privileged mode. Processes in privileged containers are
     essentially equivalent to root on the host. Defaults to false.
```

Reply:
The examples are meant to show what is not allowed, maybe that is not totally not clear.